modules/stanford_events_importer/src/Plugin/migrate/process/DateMath.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Drupal\stanford_events_importer\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Do some math on a date string.
+ *
+ * Available configuration keys:
+ * - operation: "addition"|"subtraction"
+ * - values:
+ *
+ * Examples:
+ *
+ * @code
+ * process:
+ *   plugin: stanford_events_datemath
+ *   operation: addition
+ *   source: some_date_string
+ *   values:
+ *     - another_date_string
+ *     - constants/some_number
+ * @endcode
+ *
+ * This will perform the mathematical operation on the date strings.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "stanford_events_datemath"
+ * )
+ */
+class DateMath extends ProcessPluginBase {
```

Comment:
This needs a test.

Reply:
this looks like theres gotta be an easier way to do this. :man_shrugging:

Reply:
Oh boy did I try to avoid writing this class. SMRT dates need SMRTR logic it would appear.

Reply:
@sherakama needs test.